### ISSUE TITLE
Configuration.yaml

### ISSUE THREAD

**Issue by markdcarpenter** [home-assistant/home-assistant.github.io#32395]
### Feedback

Just scratching my head when I see configuration instructions in documentation on the web quite different than instructions in my original configuration.yaml. Was there a major overhaul of the python syntax in recent releases? 

### URL

https://www.home-assistant.io/integrations/switch.mqtt/

### Version

2024.4.3

**Comment by markdcarpenter** [comment]
It was all working fine for over a year and then all of a sudden my sonoff switch didn't switch.
It was as if homeassistant could no longer interpret my configuration. So I tested the sonoff by running a browser with the ip address configured in the sonoff and it popped right up. I was able to toggle the switch on and off with no problem. But homeassistant no longer showed the switch in integrations. Restarted my router, homeassistant multiple times with no change. It's as if discovery was disabled. There was a recent update to homeassistant a couple of days ago. Maybe that broke it?
[configuration.yaml.txt](https://github.com/home-assistant/home-assistant.io/files/15049929/configuration.yaml.txt)

**Comment by markdcarpenter** [comment]
Tomorrow I will reflash Tasmota into the sonoff and test. It does return a ping, but that's the only thing left to do to fix it. Unless it became deprecated.

**Comment by jbouwh** [comment]
Platform set up has been removed for some time  now. There are 2 formats that can be used to set up manual configured MQTT items through YAML, see: https://www.home-assistant.io/integrations/mqtt/#manual-configured-mqtt-items

In the documentation we document the recommended notation. The platform setup was removed a year after we deprecated it.

I am not sure how long ago this is. If you still find examples that use an `mqtt` platform set up, they must be quite old.
